sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java

```diff
@@ -1154,6 +1154,8 @@ public static Sink sink(Schema schema) {
 
     abstract int getRowGroupSize();
 
+    abstract @Nullable Class<? extends GenericData> getAvroDataModel();
```

Comment:
Can we just return `GenericData` here?

Reply:
it needs to be serializable, since this is passed to the `DoFn` responsible for writing. The reader does the same way in the [`ReadFn`](https://github.com/apache/beam/blob/master/sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java#L1069) and [`SplitReadFn`](https://github.com/apache/beam/blob/master/sdks/java/io/parquet/src/main/java/org/apache/beam/sdk/io/parquet/ParquetIO.java#L810)

Reply:
Yes but my concern is that we have a little bit different builder's API for Read (returns `GenericData`) and Write (returns `Class<? extends GenericData>`) parts. Well, I think we can leave it at "it is" since only `withAvroDataModel(GenericData model)` is a public user's API and it's the same for both parts.

Reply:
I'll change this to be `getAvroDataModelClass` which should highlight better the differences